internal/server/iigointernal/executive.go

```diff
@@ -72,10 +64,34 @@ func (e *executive) getRuleForSpeaker() (shared.PresidentReturnContent, error) {
 
 	returnRule := e.clientPresident.PickRuleToVote(e.RulesProposals)
 
+	//Log Rule: obligation to select a rule if there is something in the proposal list
+	if e.gameState.IIGORolesBudget[shared.President]-e.gameConf.GetRuleForSpeakerActionCost >= 0 {
+		rulesProposed := len(e.RulesProposals) > 0
+
+		variablesToCache := []rules.VariableFieldName{rules.IslandsProposedRules, rules.PresidentRuleProposal}
+		valuesToCache := [][]float64{{boolToFloat(rulesProposed)}, {boolToFloat(returnRule.ActionTaken)}}
+		e.monitoring.addToCache(e.PresidentID, variablesToCache, valuesToCache)
+	}
+
 	if returnRule.ActionTaken && (returnRule.ContentType == shared.PresidentRuleProposal) {
 		if !e.incurServiceCharge(e.gameConf.GetRuleForSpeakerActionCost) {
 			return returnRule, errors.Errorf("Insufficient Budget in common Pool: getRuleForSpeaker")
 		}
+
+	}
+
+	//Log Rule: selected rule must come from rule proposal list
+	if returnRule.ActionTaken && (returnRule.ContentType == shared.PresidentRuleProposal) {
+		ruleChosenFromProposalList := false
```

Comment:
Is there a reason this isnt in the above if statement

Reply:
We log ruleChosenFromProposalList as false if it is not in the returnRule.ProposedRuleMatrix. Log it as true if it is.